### PR TITLE
James 1790 : handle attachment in emails, this PR is just a preview

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java [hunk 1]
```diff
@@ -192,11 +192,11 @@ public void run() {
                     logger.error("Exception dequeue mail", e1);
 
                 }
-            }
-        }
-        if (logger.isInfoEnabled()) {
-            logger.info("Stop " + getClass().getName() + ": " + Thread.currentThread().getName());
+            } catch (InterruptedException interrupted) {
+            	//MailSpooler is stopping
+			}
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java]: indent, but that's great :)

File: mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java [hunk 2]
```diff
@@ -42,23 +43,40 @@ public boolean isAutomaticallySent(Mail mail) throws MessagingException {
...
-                "List-Archive"})
-            .hasMoreElements();
+    	
+        return senderIsMailingList(mail)
+            || headerIsMailingList(mail);
     }
 
+	private boolean senderIsMailingList(Mail mail) {
+		MailAddress sender = mail.getSender();
+    	if (sender == null) {
```
Comment (aduprat) [inline mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java]: indent

File: mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java [hunk 3]
```diff
@@ -42,23 +43,40 @@ public boolean isAutomaticallySent(Mail mail) throws MessagingException {
...
+			|| localPart.equalsIgnoreCase("MAILER-DAEMON")
+			|| localPart.equalsIgnoreCase("LISTSERV")
+			|| localPart.equalsIgnoreCase("majordomo");
+	}
+
+
+	private boolean headerIsMailingList(Mail mail) throws MessagingException {
+		return mail.getMessage()
+			.getMatchingHeaders(
+					new String[]{
```
Comment (aduprat) [inline mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java]: You may introduce a constant for that

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/BlobId.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.base.Preconditions;
+
+public class BlobId {
+
+    public static BlobId of(String rawValue) {
+        Preconditions.checkNotNull(rawValue);
+        Preconditions.checkArgument(!rawValue.isEmpty());
```
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/BlobId.java]: Precondition.checkArgument(!Strings.isNullOrEmpty(...

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/BlobIdTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package org.apache.james.jmap.model;
```
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/BlobIdTest.java]: license

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesError.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
+package org.apache.james.jmap.model;
```
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesError.java]: license

File: server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java (hunk 1 above)
Comment (aduprat) [inline server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java]: OK

File: mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java (hunk 3 above)
Comment (aduprat) [inline mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java]: OK

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/BlobId.java (hunk 4 above)
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/BlobId.java]: OK

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/BlobIdTest.java (hunk 5 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/BlobIdTest.java]: OK

**Comment by aduprat** [comment]
#317

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesError.java (hunk 6 above)
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesError.java]: OK

**Pull request closed without merging**